src/lib/cert/x509/x509_ext.cpp

```diff
@@ -136,7 +146,7 @@ void Extensions::decode_from(BER_Decoder& from_source)
 
       m_extensions_raw.emplace(oid, std::make_pair(value, critical));
 
-      std::unique_ptr<Certificate_Extension> ext(get_extension(oid));
+      std::unique_ptr<Certificate_Extension> ext(get_extension(oid, critical));
 
       if(!ext && critical && m_throw_on_unknown_critical)
```

Comment:
At first glance I would agree, this check is now needless as validation will fail later on. However this does remove all uses of `m_throw_on_unknown_critical` from `Extensions`. This is used by CRLs and in fact I see the default parameter in `X509_CRL`s constructor has this set to `false` which seems wrong to me. Processing the CRL in the face of missing information (which a critical extension surely represents, else it wouldn't be marked critical) is probably a bad idea. So with this change, all such `m_throw_on_unknown_critical` flags in x509 (crl_ent.cpp, x509_crl.cpp) can be removed.